demo/fake_evaluation.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Wanda.Executions.FakeEvaluation do
+  @moduledoc """
+  Fake evaluation functional core.
+  """
+
+  import Wanda.Factory
+
+  @spec complete_fake_execution(
+          String.t(),
+          String.t(),
+          [Wanda.Executions.Target.t()],
+          [Wanda.Catalog.Check.t()]
+        ) :: Wanda.Executions.Result.t()
+  def complete_fake_execution(execution_id, group_id, targets, checks) do
+    result = Enum.random([:passing, :warning, :critical])
+    check_results = build_check_results_from_targets(targets, result, checks)
+
+    build(:result,
+      check_results: check_results,
+      execution_id: execution_id,
+      group_id: group_id,
+      result: result
+    )
+  end
+
+  defp build_check_results_from_targets(targets, result, checks) do
+    targets
+    |> Enum.flat_map(& &1.checks)
+    |> Enum.uniq()
+    |> Enum.map(fn check_id ->
+      check_targets = Enum.filter(targets, &(check_id in &1.checks))
+      build_check_result_from_target(check_id, check_targets, result, checks)
+    end)
+  end
+
+  defp build_check_result_from_target(check_id, check_targets, result, checks) do
+    check = Enum.find(checks, &(&1.id == check_id))
+
+    build(:check_result,
+      check_id: check_id,
+      agents_check_results:
+        Enum.map(check_targets, fn %{agent_id: agent_id} ->
+          build(:agent_check_result,
+            agent_id: agent_id,
+            facts:
+              Enum.map(check.facts, fn %{name: name} ->
```

Comment:
I still see this dot access (`checks.facts` and `checks.values`, and others around the code) here, which we could get rid off with proper pattern matching)